src/test/java/com/shapesecurity/shift/es2016/parser/expressions/BinaryExpressionTest.java

```diff
@@ -189,5 +186,7 @@ public void testBinaryExpression() throws JsError {
                 new IdentifierExpression("y")));
 
         testScript("(a, e=0)", new BinaryExpression(new IdentifierExpression("a"), BinaryOperator.Sequence, new AssignmentExpression(new AssignmentTargetIdentifier("e"), new LiteralNumericExpression(0.0))));
+
+        testScript("(-1) ** 2", new BinaryExpression(new UnaryExpression(UnaryOperator.Minus, new LiteralNumericExpression(1.0)), BinaryOperator.Exp, new LiteralNumericExpression(2.0)));
```

Comment:
Given the approach you've taken, can we also add a test `++a**2` does parse (as `(++a)**2`)?

Reply:
@bakkot Added